src/ui/webeiddialog.cpp

```diff
@@ -72,22 +72,49 @@ void fillOriginAndCertificateList(
 
 } // namespace
 
-WebEidDialog::WebEidDialog(QWidget* parent) : WebEidUI(parent), ui(new Ui::WebEidDialog)
+class WebEidDialog::Private : public Ui::WebEidDialog
+{
+public:
+    // Non-owning observer pointers.
+    QPushButton* okButton;
+    QPushButton* cancelButton;
+    QRegularExpressionValidator* pinInputValidator;
+    QTimeLine* pinTimeoutTimer;
+};
+
+WebEidDialog::WebEidDialog(QWidget* parent) : WebEidUI(parent), ui(new Private)
 {
     ui->setupUi(this);
     setWindowFlag(Qt::CustomizeWindowHint);
     setWindowFlag(Qt::WindowTitleHint);
     ui->pinLayout->setAlignment(ui->pinInput, Qt::AlignCenter);
     ui->pinInput->setAttribute(Qt::WA_MacShowFocusRect, false);
 
-    okButton = ui->buttonBox->button(QDialogButtonBox::Ok);
-    cancelButton = ui->buttonBox->button(QDialogButtonBox::Cancel);
-    makeOkButtonDefaultAndconnectSignals();
+    ui->okButton = ui->buttonBox->button(QDialogButtonBox::Ok);
+    ui->okButton->setDefault(true);
+    ui->okButton->setAutoDefault(true);
+    ui->cancelButton = ui->buttonBox->button(QDialogButtonBox::Cancel);
+    ui->cancelButton->setDefault(false);
+    ui->cancelButton->setAutoDefault(false);
+    connect(ui->cancelButton, &QPushButton::clicked, this, &WebEidDialog::rejected);
 
     // Hide PIN-related widgets by default.
     ui->pinErrorLabel->hide();
     ui->pinInput->hide();
     ui->pinEntryTimeoutProgressBar->hide();
+    ui->pinInputValidator = new QRegularExpressionValidator(ui->pinInput);
+    ui->pinInput->setValidator(ui->pinInputValidator);
+    connect(ui->pinInput, &QLineEdit::textChanged, ui->okButton,
```

Comment:
Can you please investigate this? For some reason OK button is enabled when PIN input is empty.